core/unit_test/CMakeLists.txt

```diff
@@ -293,5 +294,46 @@ KOKKOS_ADD_EXECUTABLE_AND_TEST(
   SOURCES UnitTestMain.cpp  TestHostBarrier.cpp
 )
 
+FUNCTION (KOKKOS_ADD_INCREMENTAL_TEST BACK_END_NAME)
+  KOKKOS_OPTION( ${BACK_END_NAME}_EXCLUDE_TESTS "" STRING "Incremental test exclude list" )
+  KOKKOS_OPTION( ${BACK_END_NAME}_FEATURE_LEVEL "0" STRING "Incremental test implementation level" )
+  # Add unit test main
+  SET(${BACK_END_NAME}_SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/UnitTestMainInit.cpp)
+
+  # Iterate over incremental tests in directory
+  APPEND_GLOB(INCREMENTAL_FILE_LIST ${CMAKE_CURRENT_SOURCE_DIR}/incremental/*.hpp)
```

Comment:
It just occurred to me that this (INCREMENTAL_FILE_LIST) only needs to be populated once...we should move this to before we process the experimental backend inputs and call KOKKOS_ADD_INCREMENTAL_TEST the first time.